### PR TITLE
Refactoring of extractions // Warn users for untrusted matches

### LINKED ISSUE #22: Display reverse matches with warnings (or hide them)
Also mail searches should always take priority before reverse lookups:
screenshot from 2017-01-24 13 48 39

### PR DESCRIPTION
**Pull request by ngotchac**
Closes #22 

It displays black and white info on matches that are unsafe/not trusted/verified.
Also refactored how extraction is handled to make it easier now.

It improves the overall extraction ; now it can match multiple emails/handles/etc. in a single text element.

### CODE REVIEW

File: content/components/augmentedIcon.js [hunk 1]
```diff
@@ -54,37 +54,43 @@ export default class AugmentedIcon extends Component {
...
       >
         <IdentityIcon
           address={ address }
-          className={ styles.icon }
+          className={ iconClass }
           ref={ this.handleIconRef }
           size={ height }
         />
 
         <Portal into='body'>
```
Comment (tomusdrw) [inline content/components/augmentedIcon.js:91]: In cause of unsafe accounts I would add small info text to the popup explaining what happened

**Review by tomusdrw — COMMENTED** [review]
Looks ok, just additional info about what is unsafe account would be nice.

File: content/components/augmentedIcon.js (hunk 1 above)
Comment (ngotchac) [reply content/components/augmentedIcon.js:91]: :+1:

**Comment by ngotchac** [comment]
Updated to show a warning message. Also have now better positioning.

image

**Review by tomusdrw — APPROVED** [review]

**Pull request merged**